### PR TITLE
allow kleebuilder to fail 

### PR DESCRIPTION
**Pull request by regehr**
instead of exploding and using all RAM + a testcase from building LLVM that explodes and uses all RAM without this patch

this code isn't too pretty, it takes a code path that used to always succeed and allows it to fail. but I believe it is the right answer here, we've already tried several mitigations for explosion in kleebuilder and this solves all future problems of that sort.

### CODE REVIEW

**Comment by regehr** [comment]
anyone have time to look this over? I'd like to move on with some other changes as we move towards the PLDI deadline

File: lib/Extractor/KLEEBuilder.cpp [hunk 1]
```diff
@@ -843,11 +850,14 @@ void ExprBuilder::getUBPhiPaths(Inst *I, PhiPath *Current,
     Current->BlockConstraints[I->B] = 0;
     // Continue recursively
     for (unsigned J = 0; J < Ops.size(); ++J)
-      getUBPhiPaths(Ops[J], Tmp[J], Paths, CachedPhis);
+      if (!getUBPhiPaths(Ops[J], Tmp[J], Paths, CachedPhis, Depth + 1))
+	return false;
```
Comment (chenyang78) [inline lib/Extractor/KLEEBuilder.cpp]: Two spaces?

File: lib/Extractor/KLEEBuilder.cpp [hunk 2]
```diff
@@ -843,11 +850,14 @@ void ExprBuilder::getUBPhiPaths(Inst *I, PhiPath *Current,
...
   } else {
     for (unsigned J = 0; J < Ops.size(); ++J)
-      getUBPhiPaths(Ops[J], Current, Paths, CachedPhis);
+      if (!getUBPhiPaths(Ops[J], Current, Paths, CachedPhis, Depth + 1))
+	return false;
```
Comment (chenyang78) [inline lib/Extractor/KLEEBuilder.cpp]: Same space issue?

File: lib/Extractor/Solver.cpp [hunk 3]
```diff
@@ -87,6 +89,8 @@ class BaseSolver : public Solver {
       Inst *I = IC.createVar(LHS->Width, "constant");
       InstMapping Mapping(LHS, I);
       std::string Query = BuildQuery(BPCs, PCs, Mapping, &ModelInsts, /*Negate=*/true);
+      if (Query.empty())
+	return std::make_error_code(std::errc::value_too_large);
```
Comment (chenyang78) [inline lib/Extractor/Solver.cpp]: Two spaces

File: lib/Extractor/Solver.cpp [hunk 4]
```diff
@@ -149,10 +157,11 @@ class BaseSolver : public Solver {
       }
       return EC;
     } else {
+      std::string Query = BuildQuery(BPCs, PCs, Mapping, 0);
+      if (Query.empty())
+	return std::make_error_code(std::errc::value_too_large);
```
Comment (chenyang78) [inline lib/Extractor/Solver.cpp]: same issue

File: lib/Infer/InstSynthesis.cpp [hunk 5]
```diff
@@ -214,6 +216,8 @@ std::error_code InstSynthesis::synthesize(SMTLIBSolver *SMTSolver,
       // Don't use original BPCs/PCs, they are useless
       std::string QueryStr = BuildQuery({}, LoopPCs, Mapping,
                                         &ModelInsts, /*Negate=*/true);
+      if (QueryStr.empty())
+	return std::make_error_code(std::errc::value_too_large);
```
Comment (chenyang78) [inline lib/Infer/InstSynthesis.cpp]: Ditto

File: lib/Infer/InstSynthesis.cpp [hunk 6]
```diff
@@ -268,6 +272,8 @@ std::error_code InstSynthesis::synthesize(SMTLIBSolver *SMTSolver,
       ModelVals.clear();
       InstMapping CandMapping(LHS, Cand);
       QueryStr = BuildQuery(BPCs, PCs, CandMapping, &ModelInsts, /*Negate=*/false);
+      if (QueryStr.empty())
+	return std::make_error_code(std::errc::value_too_large);
```
Comment (chenyang78) [inline lib/Infer/InstSynthesis.cpp]: Same

File: lib/Extractor/KLEEBuilder.cpp (hunk 1 above)
Comment (regehr) [inline lib/Extractor/KLEEBuilder.cpp]: Thanks... . I think emacs on my laptop has something configured badly...

**Comment by regehr** [comment]
Argh some tab chars sneaked in-- should be fixed now.

**Pull request merged**